code/modules/events/antagonist_critter_ghost_respawn.dm

```diff
@@ -20,35 +21,33 @@
 		if (..())
 			return
 
-		var/input = alert(usr, "Choose the critter type?", src.name, "Random", "Custom")
-		if (!input)
-			return
-		else if (input == "Custom")
-			input = input("Enter a /mob/living/critter path", src.name, null) as null|text
-			input = get_one_match(input, "/mob/living/critter")
-		else //random
-			input = null
+		switch (alert(usr, "Choose the critter type?", src.name, "Random", "Custom"))
+			if ("Custom")
+				src.critter_type = input("Enter a /mob/living/critter path", src.name, null) as null|text
+				src.critter_type = get_one_match(src.critter_type, "/mob/living/critter")
```

Comment:
What if they mistype and `get_one_match()` returns `null`? You probably want to let them retype instead of just letting `critter_type` be `null`, which would lead to random critters.